public/cron.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+// Semaphone part is meant to prevent concurrent execution
+// I dropped the idea of using real semaphores because a lot of 
+// providers don't allow it.
+// TODO : Maybe create a semaphore class with a fallback on file locks
+// TODO : Ensure no locks remain there for long (would block cron)
+// TODO : make a javascript more clever doing periodic calls
+//        + push here a json structure with information so that javascript call does not come back too often
+//        if there are a lot of people connected
+
+// Get the semaphore
+$fp = fopen("../resources/logs/cronlock.txt", "w+");
+
+// If semaphore can not be created, exit
+if (!flock($fp, LOCK_EX)) {
+    exit;
+}
+
+define('RESTRICTED', FALSE);
+define('ROOT', dirname(__FILE__));
+
+include_once '../config/configuration.php';
+include_once '../config/appSettings.php';
+include_once '../src/core/class.autoload.php';
+
+use leantime\domain\repositories;
+use leantime\domain\services;
+
+$config = new leantime\core\config();
+$settings = new leantime\core\appSettings();
+$settings->loadSettings($config->defaultTimezone);
+
+// NEW Audit system
+$audit = new leantime\domain\repositories\audit();
+
+$LastEvent = $audit->getLastEvent('cron');
+
+// Using audit system to prevent too frequent executions
+$lastEventDate = strtotime($LastEvent['date']);
+$nowDate = time();
+$timeSince = abs($nowDate - $lastEventDate);
+if ($timeSince < 300)
+{
+    echo "Last cron execution was on ".$LastEvent['date']. " plz come back later";
+    exit;
+}
+
+// Storing audit cron event
+$audit->storeEvent("cron", "Cron started");
+
+// TODO  check if using the session class in cron is a better idea
+session_start();
+
+if(isset($config->appUrl) && $config->appUrl != ""){
+    define('BASE_URL', $config->appUrl);
+    define('CURRENT_URL', $config->appUrl.$settings->getRequestURI($config->appUrl));
+} else{
+    define('BASE_URL', $settings->getBaseURL());
+    define('CURRENT_URL', $settings->getFullURL());
+}
+
+ob_start();
+
+if ( isset($_GET['mode']))
+{
+    if ( $_GET['mode'] && $_GET['mode'] == "debug" )
```

Comment:
you've already established the first of these 2 criteria with the outer if

Reply:
Hi @funkywaddle : good catch thanks

Improvement committed : 56b0438753af6f6d44718c9faffc3e8c5e420229